### ISSUE TITLE
[Feature]: support Classifier-Free Guidance Logits processor

### ISSUE THREAD

**Issue by zhaoyinglia** [vllm-project/vllm#5245]
### 🚀 The feature, motivation and pitch

Implementation in transformers: https://github.com/huggingface/transformers/blob/6b22a8f2d8ded6b2a681e1c078b0e1abf27d045c/src/transformers/generation/logits_process.py#L2171

Hi, the Classifier-Free Guidance Logits processor need re-run the model forward process with unconditional_ids in comput_logits process,  how to make this happen in vLLM?
